docs/examples/use_cases/pytorch/efficientnet/image_classification/dali.py

```diff
@@ -64,22 +70,32 @@ def training_pipe(data_dir, interpolation, image_size, output_layout, automatic_
                                       mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
                                       std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
 
-    return output, labels
+    if labels:
+        return output, labels
+    else:
+        return output
+
 
+@pipeline_def(exec_dynamic=True)
+def validation_pipe(data_dir, interpolation, image_size, image_crop, output_layout,
+                    dali_device="gpu", rank=0, world_size=1, send_filepaths=False):
+    jpegs, labels = input_pipeline(
+        data_dir, send_filepaths, random_shuffle=False, rank=rank, world_size=world_size)
 
-@pipeline_def
-def validation_pipe(data_dir, interpolation, image_size, image_crop, output_layout, rank=0,
-                    world_size=1):
-    jpegs, label = fn.readers.file(name="Reader", file_root=data_dir, shard_id=rank,
-                                   num_shards=world_size, random_shuffle=False, pad_last_batch=True)
+    decoder_device = "mixed" if dali_device == "gpu" else "cpu"
+    images = fn.decoders.image(jpegs, device=decoder_device, output_type=types.RGB)
 
-    images = fn.decoders.image(jpegs, device="mixed", output_type=types.RGB)
+    images = fn.resize(images, resize_shorter=image_size,
+                       interp_type=interpolation, antialias=False)
 
-    images = fn.resize(images, resize_shorter=image_size, interp_type=interpolation,
-                       antialias=False)
+    images = images.gpu()
```

Comment:
Did I read it right that before the validation pipeline was returning CPU data, and now it's GPU? Why?